packages/TextEditor/src/TextEditor.js

```diff
@@ -67,16 +73,20 @@ export default function TextEditor(props) {
 
   const initExtended = (tiny?.props && tiny.props?.init) || {};
 
+  React.useEffect(() => {
+    setNextIdKey(prev => prev + 1);
+  }, [kind]);
+
   return (
     <Editor
+      key={idKey}
```

Comment:
Wont the `key={kind}` be enough, though? 
It will re-mount the component if `kind` changes from SIMPLE to ADVANCE, and vice-versa

Reply:
yeap true, but at this moment don't know if we want to also check if `init` from `<TextEditor.Tiny />` should change as well, or any other prop where the TextEditor needs to rerender. 

So if that is the case is easier to add a dependency on the next key and will keep working and escalating easily.

Reply:
@nahumzs If you add `init` as a dependency it will probably trigger the `effect` all the time, unless it is wrapped in memo (and even then it is no a guarantee)

I'd go with the simplest solution that solves the task right now `key={kind}` and then will expand it later if required. If requirements change, you'll have to update `TextEditor.js` anyway

Reply:
Generally, I try to avoid using effect unless it is the only option to solve the task (e.g. when you need to send an HTTP request)

Reply:
k, I think in this case useEffect is the only solution. unless we don't allow switching between kinds, but I think we should.

Reply:
k, lets go with the simplest solution for now then